pb/paste/model.py

```diff
@@ -38,12 +38,9 @@ def _put(stream):
     b = stream.read()
     digest = sha1(b).hexdigest()
     size = len(b)
-    try:
-        if stream.getbuffer().nbytes > 2 ** 23:
-            b = get_fs().put(b)
-    except AttributeError:
-        # FIXME: what the actual fuck, mitsuhiko?
-        b = get_fs().put(b)
```

Comment:
This attribute error happens because (in some version of werkzeug/flask, which may or may not include the latest version) the response object did not provide a nbytes attribute. There are a few ways to deal with this, this implementation being the laziest option while still being arguably correct.

Reply:
So when the  `AttributeError` occurs will always store in gridfs even if the file is smaller than 8MiB?

Reply:
Correct.

Reply:
The specific `AttributeError` may be

```
AttributeError: 'SpooledTemporaryFile' object has no attribute 'getbuffer'
```

I think the reason can be seen here: https://github.com/pallets/werkzeug/issues/1344

Unfortunately, after this PR merged, a new error will occur:
```
AttributeError: 'SpooledTemporaryFile' object has no attribute 'getvalue'
```
---

@buhman Why not use `size` directly? 😢

```py
b = stream.read()
digest = sha1(b).hexdigest()
size = len(b)

if size > 2 ** 23:
    b = get_fs().put(b)
```

Reply:
Heh, if Werkzeug broke their API again, I suppose this needs to either update or pin an older Werkzeug.

Does 0.14.1 not work? https://github.com/ptpb/pb/blob/master/requirements.txt#L2

Reply:
@buhman 

From https://github.com/pallets/werkzeug/issues/1344#issuecomment-440589308 we can know, werkzeug 0.12 is ok and then `SpooledTemporaryFile` have been added since 0.13, so this `AttributeError` occurs.

In this PR https://github.com/pallets/werkzeug/pull/1409 , try to solve some problems through patch, the version is 0.15. But even with the latest 0.16 (not break the big version), current code doesn't work either.

I tried it based on the above PR 1409 and it worked fine with a simple modification:
![image](https://user-images.githubusercontent.com/13994362/157043843-a37a558b-65c5-450a-83d1-423590377892.png)

However, I still want to ask the question mentioned above again, why not use `size` directly? 😕️ Looking forward to your answer.